### PR TITLE
Update azure sdk to latest version

### PR DESCRIPTION
**Pull request by sandeepkru**
Azure SDK latest version fixes couple of bug fixes and adds transactional MD5 to stageblock API

### CODE REVIEW

File: vendor/modules.txt [hunk 1]
```diff
@@ -6,7 +6,7 @@ bazil.org/fuse/fuseutil
 cloud.google.com/go/compute/metadata
 # github.com/Azure/azure-pipeline-go v0.0.0-20180607212504-7571e8eb0876
 github.com/Azure/azure-pipeline-go/pipeline
-# github.com/Azure/azure-storage-blob-go v0.0.0-20180712005634-eaae161d9d5e
+# github.com/Azure/azure-storage-blob-go v0.0.0-20180906215025-bb46532f68b7
```
Comment (ncw) [inline vendor/modules.txt]: I see there is an official 0.2.0 release: https://github.com/Azure/azure-storage-blob-go/releases released 15 hours ago!

Do you think it is worth updating to that?

Comment (sandeepkru) [reply vendor/modules.txt]: Agree, that was my intention, will pull the right version.

**Comment by ncw** [comment]
The diff of the go.mod still looks odd.  There are some strange extra dependencies and the version is still 0.0.0.  I have to say I'm still learning `go mod` so not sure why that is!

**Comment by sandeepkru** [comment]
> The diff of the go.mod still looks odd. There are some strange extra dependencies and the version is still 0.0.0. I have to say I'm still learning go mod so not sure why that is!

I'm not yet successful pulling `v0.2.0` , I tried `go get package@version` and `go get package@sha1`, neither of those are successful, and I even tried `go mod edit -replace=old[@v]=new[@v] -dropreplace=old[@v]` no luck with this either. I'm also still trying to figure out `go mod`, not sure about additional strange dependencies.

**Comment by ncw** [comment]
Well if you think you've got the new code then we should merge this as-is and worry about go mod later!

**Comment by ncw** [comment]
OK. I'm happy for you to merge this when you are satisfied :-)

**Comment by sandeepkru** [comment]
`go.mod` has some updates related to `tools` like `lint, mod, errcheck`. I don't know why,  as far as azure changes concerned, it pull all changes as part of `v.0.2.0` which is the `tip` today and locally tested changes, I believe its safe to merge.

**Comment by ncw** [comment]
OK I'll merge this now then :-)

**Pull request merged**